src/Context/ContextProviderInterface.php

```diff
@@ -37,7 +37,7 @@ public function setProvidedValue($name, $value);
    * @param string $name
    *   The name of the provided context in the plugin definition.
    *
-   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   * @throws \Drupal\rules\Exception\ContextUndefinedInvalidException
```

Comment:
I think we should not touch the exceptions that are already fully namespaced. This patch should only replace the global exceptions that don't have a namespace.